openmc/data/effective_dose/dose.py

```diff
@@ -2,40 +2,55 @@
 
 import numpy as np
 
-_FILES = (
-    ('electron', 'electrons.txt'),
-    ('helium', 'helium_ions.txt'),
-    ('mu-', 'negative_muons.txt'),
-    ('pi-', 'negative_pions.txt'),
-    ('neutron', 'neutrons.txt'),
-    ('photon', 'photons.txt'),
-    ('photon kerma', 'photons_kerma.txt'),
-    ('mu+', 'positive_muons.txt'),
-    ('pi+', 'positive_pions.txt'),
-    ('positron', 'positrons.txt'),
-    ('proton', 'protons.txt')
-)
-
-_DOSE_ICRP116 = {}
-
-
-def _load_dose_icrp116():
-    """Load effective dose tables from text files"""
-    for particle, filename in _FILES:
-        path = Path(__file__).parent / filename
-        data = np.loadtxt(path, skiprows=3, encoding='utf-8')
-        data[:, 0] *= 1e6   # Change energies to eV
-        _DOSE_ICRP116[particle] = data
-
-
-def dose_coefficients(particle, geometry='AP'):
-    """Return effective dose conversion coefficients from ICRP-116
+import openmc.checkvalue as cv
+
+
+_FILES = {
+    ('icrp116', 'electron'): Path('icrp116') / 'electrons.txt',
+    ('icrp116', 'helium'): Path('icrp116') / 'helium_ions.txt',
+    ('icrp116', 'mu-'): Path('icrp116') / 'negative_muons.txt',
+    ('icrp116', 'pi-'): Path('icrp116') / 'negative_pions.txt',
+    ('icrp116', 'neutron'): Path('icrp116') / 'neutrons.txt',
+    ('icrp116', 'photon'): Path('icrp116') / 'photons.txt',
+    ('icrp116', 'photon kerma'): Path('icrp116') / 'photons_kerma.txt',
+    ('icrp116', 'mu+'): Path('icrp116') / 'positive_muons.txt',
+    ('icrp116', 'pi+'): Path('icrp116') / 'positive_pions.txt',
+    ('icrp116', 'positron'): Path('icrp116') / 'positrons.txt',
+    ('icrp116', 'proton'): Path('icrp116') / 'protons.txt',
+    ('icrp119', 'neutron'): Path('icrp119') / 'neutrons.txt',
+}
+
+_DOSE_TABLES = {key: None for key in _FILES.keys()}
+
+
+def _load_dose(particle, library='icrp116'):
+    """Load effective dose tables from text files
+
+    Parameters
+    ----------
+    particle : {'neutron', 'photon', 'photon kerma', 'electron', 'positron'}
+        Incident particle
+    library : {'icrp116', 'icrp119'}
+        The dose conversion library to use
+
+    """
+    print(f'loading {particle} {library}')
+    path = Path(__file__).parent / _FILES[library, particle]
+    data = np.loadtxt(path, skiprows=3, encoding='utf-8')
+    data[:, 0] *= 1e6   # Change energies to eV
+    _DOSE_TABLES[library, particle] = data
+
+
+def dose_coefficients(particle, geometry='AP', library='icrp116'):
```

Comment:
I feel like "library" is already an overloaded term. How about "data_source" instead?

Reply:
data_source is good with me, I have changed the code accordingly.